### PR TITLE
Fix #3476

### PR DESCRIPTION
**Pull request by rana-nouman**
**Description**:
The goal of this task is to implement unit tests for the TokenController in the Gaia project. The TokenController is responsible for handling OAuth token generation, token expiry, remember me functionality and session validity. The following unit tests need to be implemented:

1. **testPostActionValidRequest:**
Verify that a valid OAuth request returns a 200 status code.

2. **testTokenExpiry:**
Ensure that an expired token results in an unauthorized exception.

3. **testSetRememberMe:**
Check that the remember me functionality sets the correct expiration time.

4. **testRememberMeExpiry:**
Verify that the remember me functionality expires correctly.

5. **testSessionValidity:**
Confirm that the session validity is set correctly.

6. **testSessionExpiry:**
Test the session expiry functionality by mocking the OAuth request, setting a custom session timeout, and attempting to refresh the access token.

**Acceptance Criteria:**
- The test suite should pass without errors.
- All tests should be isolated and independent; external dependencies should be mocked as needed.
- Code readability and maintainability should be improved.

**Note:** 
While working with this, New issue of type **_bug_** detected.
http://prom.projects4.me/app/project/cb4567905acd-06d6-1141-7dd99b59dd21/issue/3478

### CODE REVIEW

File: tests/api/controllers/TokenControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+     *
+     * @return void
+     */
+    public function testSetRememberMe(): void
+    {
+        $this->mockOAuthRequest(true);
+        $this->mockOAuthServer();
+        $this->getAccessToken();
+        $user = \Gaia\MVC\Models\User::findFirstByUsername('testUserOAuth');
+        $this->assertGreaterThan(gmdate('Y-m-d H:i:s'), $user->rememberMe);
```
Comment (gollomer) [inline tests/api/controllers/TokenControllerTest.php]: Shouldn't this be GreaterThanEqual CurrentDate+config(rememberMe Timeout)?

File: tests/api/controllers/TokenControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+     *
+     * @return void
+     */
+    public function testRememberMeExpiry(): void
+    {
+        $this->mockOAuthRequest(false);
+        $this->mockOAuthServer();
+        $this->getAccessToken();
+        $user = \Gaia\MVC\Models\User::findFirstByUsername('testUserOAuth');
+        $this->assertLessThan(gmdate('Y-m-d H:i:s'), $user->rememberMe);
```
Comment (gollomer) [inline tests/api/controllers/TokenControllerTest.php]: Same goes here..

**Review by gollomer — CHANGES_REQUESTED** [review]

File: tests/api/controllers/TokenControllerTest.php (hunk 1 above)
Comment (rana-nouman) [reply tests/api/controllers/TokenControllerTest.php]: Test case updated with the instructions. Please review.

File: tests/api/controllers/TokenControllerTest.php (hunk 2 above)
Comment (rana-nouman) [reply tests/api/controllers/TokenControllerTest.php]: Test case updated with the instructions. Please review.

**Review by gollomer — APPROVED** [review]

**Pull request merged**